lib/fastifySession.js

```diff
@@ -145,7 +145,7 @@ function onSend (options) {
 
     if (!shouldSaveSession(request, options.cookie, options.saveUninitialized)) {
       // if a session cookie is set, but has a different ID, clear it
-      if (request.cookies[options.cookieName] && request.cookies[options.cookieName] !== session.encryptedSessionId) {
+      if (request.cookies?.[options.cookieName] !== session.encryptedSessionId) {
```

Comment:
Here should not be changed. Optional Chaining is not identical to what it is checking here.

```suggestion
      if (request.cookies[options.cookieName] && request.cookies[options.cookieName] !== session.encryptedSessionId) {
```